gdal/ogr/ogrsf_frmts/shape/shpopen.c

```diff
@@ -504,21 +497,21 @@ SHPOpen( const char * pszLayer, const char * pszAccess )
 /*                                                                      */
 /*      Open the .shp and .shx files based on the basename of the       */
 /*      files or either file name.                                      */
+/*      There are 2 types of SHPOpen() : with .shx restoring code       */
+/*      (SHPOpenLLEx) and normal one.                                   */
```

Comment:
There seems to be a lot of code duplication between both functions. Why not just having the extending one taking an extra boolean argument to restore the .shx and the existing one to call it with FALSE ?